### PR TITLE
check type before setting to object

### PR DESCRIPTION
**Pull request by Frees**

### CODE REVIEW

**Comment by RobertoSchneiders** [comment]
@Frees can you write a test for this problem?

**Comment by Frees** [comment]
unit TestRestJsonUtils;
{

  Delphi DUnit Test Case
  ----------------------
  This unit contains a skeleton test case class generated by the Test Case Wizard.
  Modify the generated code to correctly setup and call the methods from the unit 
  being tested.

}

interface

uses
  TestFramework, DateUtils, SysUtils, RestJsonGenerics, RestJsonUtils, TypInfo, System.Generics.Collections;

type

  TestTJsonUtil = class(TTestCase)
  strict private
  published
    procedure TestUnMarshal;
  end;

  TSubItems = class
    Price: Currency;
    Quality: Integer;
  end;

  TItems = class
    ItemId: string;
    SubItems: TObjectList<TSubItems>;
  end;

implementation

procedure TestTJsonUtil.TestUnMarshal;
var
  s: string;
  SearchResult: TObjectList<TItems>;
begin
  s :=
    '[{' + sLineBreak +
    '  "SubItems": [{' + sLineBreak +
    '    "Quality": null,' + sLineBreak +
    '    "Price": 1234.56' + sLineBreak +
    '  }],' + sLineBreak +
    '  "ItemId": null' + sLineBreak +
    '}]';

  SearchResult := TJsonUtil.UnMarshal<TObjectList<TItems>>(s);
  CheckTrue(SearchResult <> nil);
  CheckTrue(SearchResult.Count = 1);
  CheckTrue(SearchResult[0].SubItems.Count = 1);
  CheckEquals(1234.56, SearchResult[0].SubItems[0].Price, 0.001);
  CheckEquals(0, SearchResult[0].SubItems[0].Quality);

  SearchResult.OwnsObjects := true;
  SearchResult.Free;
end;

initialization
  RegisterTest(TestTJsonUtil.Suite);
end.

**Comment by RobertoSchneiders** [comment]
We already have a test class for `DBXJsonUnMarshal.pas`, wich is `TestDbxJsonUnMarshal.pas`. It would be better to put this test with the others. Don't you think? 

You have some changes on the `DelphiXE3_RestApiTests.dproj` file, are you sure that's necessary?

**Comment by Frees** [comment]
I commited test for my change

File: unittest/TestDbxJsonUnMarshal.pas [hunk 1]
```diff
@@ -321,6 +327,15 @@ procedure TTestDbxJsonUnMarshal.valueList;
...
 end;
 
+procedure TTestDbxJsonUnMarshal.valueNull;
+var
+  Items: TTestNullValue;
+begin
+  Items := TDbxJsonUnMarshal.FromJson<TTestNullValue>('{"nullValue": null,"floatValue": 1234.56}');
+  CheckNotNull(Items);
+  Items.Free;
+end;
```
Comment (RobertoSchneiders) [inline unittest/TestDbxJsonUnMarshal.pas]: What are the expected values of `Items.floatValue` and `Items.nullValue`?

**Comment by Frees** [comment]
I add check for expected values

**Comment by RobertoSchneiders** [comment]
:+1:  
Thanks @Frees.

**Pull request merged**